### ISSUE TITLE
super() must not be called many times

### ISSUE THREAD

**Issue by falsandtru** [Microsoft/TypeScript#23422]
Calling super() twice makes a runtime error.

**TypeScript Version:**  2.7.0-dev.201xxxxx

**Search Terms:** 

**Code**

```ts
class C extends Object {
  constructor() {
    super();
    super();
  }
}
```

**Expected behavior:**

error

**Actual behavior:**

pass

**Playground Link:**  

**Related Issues:**

**Comment by JoshuaKGoldberg** [comment]
See discussion in https://github.com/palantir/tslint/issues/1983

**Comment by falsandtru** [comment]
It seems not standardized and still invalid now.

**Comment by mhegazy** [comment]
Not sure we can do more much here. In the general case the compiler can not gurantee how many times the method will be called. The check we can add is that the super call does not appear in a block more than once; first we do not have the infrastructure to do that, and second I think there will be false positives with conditions that the user know are mutually exclusive. This seems like a linter check to me.

**Comment by falsandtru** [comment]
Today, compiler seems to be changed to allow putting code before super call. So counting super calls is certainly difficult.

```ts
class C extends Object {
  constructor() {
    0
    super();
  }
}
```
